### PR TITLE
Support MobileViT

### PR DESCRIPTION
**Pull request by yingfhu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcls/models/backbones/mobilevit.py [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+            x = self.conv_fusion(torch.cat((shortcut, x), dim=1))
+        return x
+
+
+@MODELS.register_module()
+class MobileViT(BaseBackbone):
+    """MobileViT backbone.
+
+    Args:
+        widen_factor (float): Width multiplier, multiply number of
```
Comment (mzr1996) [inline mmcls/models/backbones/mobilevit.py:156]: Please fix the docstring.

File: mmcls/models/backbones/mobilevit.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+from torch import nn
+
+from .base_backbone import BaseBackbone
+from .mobilenet_v2 import InvertedResidual
+from .vision_transformer import TransformerEncoderLayer
+
+
+class MobileVitBlock(nn.Module):
+    """ MobileViT block
+        Paper: https://arxiv.org/abs/2110.02178?context=cs.LG
```
Comment (mzr1996) [inline mmcls/models/backbones/mobilevit.py:18]: Please move the paper link to the backbone docstring and use link format.

**Review by mzr1996 — CHANGES_REQUESTED** [review]

File: configs/mobilevit/metafile.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      Training Data: ImageNet-1k
+      Architecture:
+        - MobileViT Block
+    Paper:
+      URL: https://arxiv.org/abs/2110.02178
+      Title: MobileViT Light-weight, General-purpose, and Mobile-friendly Vision Transformer
+    README: configs/mobilevit/README.md
+
+Models:
+  - Name: mobilevit_small_in1k
```
Comment (mzr1996) [inline configs/mobilevit/metafile.yml:13]: If the weight is converted from the official repo, it should be like `mobilevit-small_3rdparty_in1k`

File: configs/mobilevit/mobilevit-small_8xb128_in1k.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+_base_ = [
+    '../_base_/models/mobilevit/mobilevit_s.py',
+    '../_base_/datasets/imagenet_bs32.py',
+    '../_base_/default_runtime.py',
+    '../_base_/schedules/imagenet_bs256.py',
+]
+
+# no normalize for original implements
```
Comment (mzr1996) [inline configs/mobilevit/mobilevit-small_8xb128_in1k.py:8]: since no normalize, why the std is 255 instead of 1.

File: configs/mobilevit/mobilevit-small_8xb128_in1k.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+# use bgr directly
```
Comment (mzr1996) [inline configs/mobilevit/mobilevit-small_8xb128_in1k.py:9]: move this line above the `to_rgb=False`.

**Review by mzr1996 — CHANGES_REQUESTED** [review]

File: configs/mobilevit/mobilevit-small_8xb128_in1k.py (hunk 5 above)
Comment (yingfhu) [reply configs/mobilevit/mobilevit-small_8xb128_in1k.py:9]: done

File: configs/mobilevit/metafile.yml (hunk 3 above)
Comment (yingfhu) [reply configs/mobilevit/metafile.yml:13]: done

File: mmcls/models/backbones/mobilevit.py (hunk 1 above)
Comment (yingfhu) [reply mmcls/models/backbones/mobilevit.py:156]: fixed

File: mmcls/models/backbones/mobilevit.py (hunk 2 above)
Comment (yingfhu) [reply mmcls/models/backbones/mobilevit.py:18]: done

File: configs/mobilevit/mobilevit-small_8xb128_in1k.py (hunk 4 above)
Comment (yingfhu) [reply configs/mobilevit/mobilevit-small_8xb128_in1k.py:8]: std=1 all pixels fall into range 0-255， in original repo，input should be in range 0-1

**Review by mzr1996 — APPROVED** [review]
